src/Agent/NewRelic/Agent/Core/DataTransport/DataStreamingService.cs

```diff
@@ -40,30 +42,29 @@ public ResponseStreamWrapper(int consumerID, IAsyncStreamReader<TResponse> respo
 
         private async Task<int> WaitForResponse()
         {
-            bool success;
+            var success = false;
             try
             {
                 success = await _responseStream.MoveNext(_streamCancellationToken);
             }
-            catch (Exception ex)
+            catch (RpcException rpcEx)
```

Comment:
Confirmed that `success = await _responseStream.MoveNext(_streamCancellationToken)` will never result an `AggregateException`. The `await` mechanism implicitly unwraps propagated AggregateException, always leaving the underlying exception. 👍 on this catch.
LGTM!